packages/core/src/provider.ts

```diff
@@ -445,20 +445,23 @@ export function switchChain(
 
 export function addNewChain(
   provider: EIP1193Provider,
-  chain: Chain
+  chain: Chain,
+  rpcUrl?: string,
+  label?: string,
+  token?: string,
 ): Promise<unknown> {
   return provider.request({
     method: 'wallet_addEthereumChain',
     params: [
       {
         chainId: chain.id,
-        chainName: chain.label,
+        chainName: chain.label || label,
```

Comment:
What happens if label, token or rpcUrl aren't defined on init and one or any subset are missing when this is called?

Reply:
In theory, the call should fail, and the provider should throw an implementation-dependent error. Not sure what will happen in practice, and how it's handled here...

Reply:
correct- the call will fail. We currently throw and error and display this message: 

<img width="945" alt="Screenshot 2023-03-15 at 4 54 02 PM" src="https://user-images.githubusercontent.com/33187102/225461155-e5315cd5-2f34-4f51-bdb2-c6bee6f587c8.png">

My understanding is that it would be up to the dapp to handle?

Reply:
Yep. This modal is good enough for the case where the app doesn't care - if it does care, I think the whole `addChain` invocation logic can be implemented on the app side - to then follow with the onboard switch chain call. This is not the most straightforward way around the use of the available API for the users, but a good one for those who know what they're doing. We might want to note it in the docs though.